webapp/views.py

```diff
@@ -19,6 +30,197 @@ def get_cve(cve_id):
     return cve
 
 
+@use_kwargs(CVEsParameters, location="query")
+@marshal_with(CVEsAPISchema, code=200)
+def get_cves(**kwargs):
+    query = kwargs.get("q", "").strip()
+    priority = kwargs.get("priority")
+    package = kwargs.get("package")
+    limit = kwargs.get("limit", 20)
+    offset = kwargs.get("offset", 0)
+    component = kwargs.get("component")
+    versions = kwargs.get("version")
+    statuses = kwargs.get("status")
+    order_by = kwargs.get("order")
+
+    releases_query = db_session.query(Release).order_by(Release.release_date)
+
+    if versions and not any(a in ["", "current"] for a in versions):
+        releases_query = releases_query.filter(Release.codename.in_(versions))
+    else:
+        releases_query = releases_query.filter(
+            or_(
+                Release.support_expires > datetime.now(),
+                Release.esm_expires > datetime.now(),
+            )
+        ).filter(Release.codename != "upstream")
+
+    releases = releases_query.all()
+
+    should_filter_by_version_and_status = (
+        versions and statuses and len(versions) == len(statuses)
+    )
+
+    clean_versions = []
+    clean_statuses = []
+    if should_filter_by_version_and_status:
+        raw_all_statuses = db_session.execute(
+            "SELECT unnest(enum_range(NULL::statuses));"
+        ).fetchall()
+        all_statuses = ["".join(s) for s in raw_all_statuses]
+
+        clean_versions = [
```

Comment:
I like list comprehensions, but here it gets quite difficult to read. I would rather go with a for loop
```py
clean_versions = []
for version in versions:
  if version not in ["", "current"]:
     clean_versions.append([versions])
  else:
     clean_versions.append([r.codename for r in releases])
```

Reply:
I changed this for both versions and statuses.